ReactiveXComponent/RabbitMq/RabbitMqSubscriber.cs

```diff
@@ -0,0 +1,238 @@
+            key => new ThreadSafeList<Action<MessageEventArgs>> { stateMachineListener },
+            (key, oldValue) =>
+            {
+                oldValue.Add(stateMachineListener);
+                return oldValue;
+            });
+        }
+
+        public event EventHandler<string> ConnectionFailed;
+
+        private void ChannelOnModelShutdown(object sender, ShutdownEventArgs shutdownEventArgs)
+        {
+            ConnectionFailed?.Invoke(this, shutdownEventArgs.ReplyText);
+        }
+
+        private void DeleteSubscription(SubscriberKey subscriberkey)
+        {
+            RabbitMqSubscriberInfos rabbitMqSubscriberInfos;
+            _subscribers.TryRemove(subscriberkey, out rabbitMqSubscriberInfos);
+            if (rabbitMqSubscriberInfos == null) return;
+            rabbitMqSubscriberInfos.Channel.ModelShutdown -= ChannelOnModelShutdown;
+            rabbitMqSubscriberInfos.Subscriber.OnCancel();
+            rabbitMqSubscriberInfos.Channel.Close();
+        }
+
+        private void DispatchMessage(BasicDeliverEventArgs basicAckEventArgs)
+        {
+            var obj = _serializer.Deserialize(new MemoryStream(basicAckEventArgs.Body));
+            var msgEventArgs = new MessageEventArgs(
+                RabbitMqHeaderConverter.ConvertStateMachineRef(basicAckEventArgs.BasicProperties.Headers),
+                obj);
+            OnMessageReceived(msgEventArgs);
+        }
+
+        private void OnMessageReceived(MessageEventArgs e)
+        {
+            MessageReceived?.Invoke(this, e);
+        }
+
+        public void Unsubscribe(string component, string stateMachine, Action<MessageEventArgs> stateMachineListener)
+        {
+            var subscriberKey = new SubscriberKey(_xcConfiguration.GetComponentCode(component), _xcConfiguration.GetStateMachineCode(component, stateMachine));
+            DeleteSubscription(subscriberKey);
+            ThreadSafeList<Action<MessageEventArgs>> listStateMachineListener;
+            _listenerToUpdateBySubscriberKeyRepo.TryGetValue(subscriberKey, out listStateMachineListener);
+            if (listStateMachineListener == null || !listStateMachineListener.Contains(stateMachineListener)) return;
+                listStateMachineListener.Remove(stateMachineListener);
+        }
+
+        private void Close()
+        {
+            foreach (var subscriberkey in _listenerToUpdateBySubscriberKeyRepo.Keys)
+            {
+                DeleteSubscription(subscriberkey);
+            }
+            _listenerToUpdateBySubscriberKeyRepo.Clear();
+            _connection.Close();
+        }
+
+        private void Dispose(bool disposing)
+        {
+            if (_disposed)
+                return;
+
+            if (disposing)
+            {
+                Close();
+            }
+            _disposed = true;
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
The call to GC.SuppressFinalize() is useless as you do not implement a destructor (finalizer) for your class.